src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
+
+            foreach (TSource item in source)
+            {
+                if (queue.Count == count)
```

Comment:
It seems to me that there are three phases here, with something we can ignore in each.

1. We are filling up the buffer until we reach `count`. We can ignore yielding and just load in. We also don't have to read the `Count` property because we can decrement our local `count` until it hits zero.
2. We are receiving yet more items. As long as `MoveNext()` returns true we can `Dequeue` because we know we'll have a matching `Enqueue`. No need to pay attention to the count at all any more.
3. We got a false from `MoveNext()`. At this point we can quit, and ignore the contents of the queue.